src/views/ThreeWProjectForm/index.tsx

```diff
@@ -0,0 +1,1004 @@
+                // FIXME: event dtype id is a must inside event mini but
+                // its not defined under event_detail of this response
+                dtype: { id: response.event_detail?.dtype } as EventItem['dtype'],
+            }]);
+            setValue({
+                ...response,
+                // Set beginning is_annual_report switch according to
+                // the filled-in annual split details
+                is_annual_report: response.annual_split_detail?.length >= 1,
+                annual_split_detail: response.annual_split_detail.map((split) => ({
+                    ...injectClientId(split),
+                })),
+            });
+        },
+    });
+
+    const {
+        deployments_project_status: projectStatusOptions,
+        deployments_project_operation_type: operationTypeOptions,
+        deployments_project_programme_type: programmeTypeOptions,
+        api_visibility_char_choices: visibilityTypeOptions,
+    } = useContext(ServerEnumsContext);
+
+    const projectStatusOptionsMap = listToMap(
+        projectStatusOptions,
+        (d) => d.key,
+        (d) => d.value,
+    );
+
+    const error = getErrorObject(formError);
+    const handleProjectCountryChange = useCallback(
+        (val: number | undefined, name: 'project_country') => {
+            setFieldValue(val, name);
+            setFieldValue(undefined, 'project_districts' as const);
+            // FIXME: Add admin2 select input
+        },
+        [setFieldValue],
+    );
+
+    const isTotalRequired = value.status === PROJECT_STATUS_COMPLETED;
+
+    const shouldDisableTotalTarget = !isFalsy(value.target_male)
+        || !isFalsy(value.target_female)
+        || !isFalsy(value.target_other);
+
+    const shouldDisableTotalReached = !isFalsy(value.reached_male)
+        || !isFalsy(value.reached_female)
+        || !isFalsy(value.reached_other);
+
+    const {
+        pending: fetchingPrimarySectors,
+        response: primarySectorOptions,
+    } = useRequest({
+        url: '/api/v2/primarysector',
+    });
+
+    const {
+        pending: pendingMe,
+        response: meResponse,
+    } = useRequest({
+        url: '/api/v2/user/me/',
+    });
+
+    const visibilityOptions = useMemo(() => {
+        if (!visibilityTypeOptions || !meResponse) {
+            return undefined;
+        }
+        let projectVisibilityOptions = [...visibilityTypeOptions];
+        if (meResponse?.profile?.org_type === 'OTHR') {
+            projectVisibilityOptions = projectVisibilityOptions.filter((x) => x.key === 'public' || x.key === 'logged_in_user');
+        } else if (meResponse?.profile?.org_type === 'NTLS') {
+            projectVisibilityOptions = projectVisibilityOptions.filter(
+                (x) => x.key === 'public' || x.key === 'logged_in_user' || x.key === 'ifrc_ns',
+            );
+        }
+
+        return projectVisibilityOptions;
+    }, [
+        visibilityTypeOptions,
+        meResponse,
+    ]);
```

Comment:
Let's discuss on the visibility options with @frozenhelium 
Is this visibility filtering global to go or just this form?
Also, is this also checked on the server?